packages/block-library/src/navigation/index.php

```diff
@@ -165,7 +177,8 @@ function build_navigation_html( $block, $colors, $font_sizes ) {
 		$html .= '</a>';
 		// End anchor tag content.
 
-		if ( count( (array) $block['innerBlocks'] ) > 0 ) {
+		// Only include innerBlocks that have a label set in the count. Otherwise a submenu with no links will be rendered.
+		if ( count( (array) array_filter( $block['innerBlocks'], 'has_navigation_label' ) ) > 0 ) {
```

Comment:
This is getting pretty involved. Maybe we could save that in a variable to make that conditional easier to grok?